integration-tests/cli/product-json-to-xlsx.it.js

```diff
@@ -0,0 +1,885 @@
+            const categories = 'Parent Category>child Category'
+            expect(product[0]).toEqual(expect.objectContaining({ categories }))
+            expect(product[1]).toEqual(expect.objectContaining({ categories }))
+            expect(product[2]).toEqual(expect.objectContaining({ categories }))
+          })
+
+          it('contains resolved `state`', () => {
+            const state = 'stateKey'
+            expect(product[0]).toEqual(expect.objectContaining({ state }))
+            expect(product[1]).toEqual(expect.objectContaining({ state }))
+            expect(product[2]).toEqual(expect.objectContaining({ state }))
+          })
+
+          it('contains resolved `taxCategory`', () => {
+            const tax = 'new-tax-category'
+            expect(product[0]).toEqual(expect.objectContaining({ tax }))
+            expect(product[1]).toEqual(expect.objectContaining({ tax }))
+            expect(product[2]).toEqual(expect.objectContaining({ tax }))
+          })
+
+          it('contains variants `SKUs`', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({ sku: 'M00FCKV' })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({ sku: 'M00FCKV-11' })
+            )
+            expect(product[2]).toEqual(
+              expect.objectContaining({ sku: 'M00FCKV-22' })
+            )
+          })
+
+          it('contains variants `keys`', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({ variantKey: 'master-var-1' })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({ variantKey: 'normal-var-1' })
+            )
+            expect(product[2]).toEqual(
+              expect.objectContaining({ variantKey: 'normal-var-2' })
+            )
+          })
+
+          // Test for custom attributes
+          it('contains variants custom attributes', () => {
+            expect(product[0]).toEqual(
+              expect.objectContaining({ 'text-attribute': 'Master Var attr' })
+            )
+            expect(product[1]).toEqual(
+              expect.objectContaining({ 'text-attribute': 'First Var attr' })
+            )
+            expect(product[2]).toEqual(
+              expect.objectContaining({ 'text-attribute': 'Second Var attr' })
+            )
+          })
+        })
+      })
+
+      describe('WITH HEADERS::should write products to `XLSX` file', () => {
+        let xlsxFile
+        let excel
+        let products
+        const templateFile = path.join(__dirname, '/helpers/product-headers.csv')
+
+        beforeAll(async () => {
+          xlsxFile = tmp.fileSync({ postfix: '.xlsx' }).name
+          const cmd = `${binPath} -p ${projectKey} -i ${productsJsonFile} -t ${templateFile} --referenceCategoryBy name -o ${xlsxFile}`
+          const [stdout, stderr] = await exec(cmd)
+          expect(stdout).toBeTruthy()
```

Comment:
assertions inside the `beforeAll`?

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper